app/renderer/components/navigation/urlBarSuggestions.js

```diff
@@ -3,112 +3,103 @@
         const currentIndex = index + i
-        const selected = this.activeIndex === currentIndex || (!this.activeIndex && currentIndex === 0 && this.props.hasSuggestionMatch)
+        const selected = this.props.activeIndex === currentIndex || (this.props.activeIndex == null && currentIndex === 0 && this.props.hasSuggestionMatch)
         return <UrlBarSuggestionItem
           suggestion={suggestion}
           selected={selected}
           currentIndex={currentIndex}
-          i={i}
-          onMouseOver={this.onMouseOver}
-          onClick={this.onSuggestionClicked} />
+          i={i} />
       }))
-      index += suggestions.size
+      index += suggestions.length
     }
-    addToItems(historySuggestions, 'historyTitle', locale.translation('historySuggestionTitle'), 'fa-clock-o')
-    addToItems(bookmarkSuggestions, 'bookmarksTitle', locale.translation('bookmarksSuggestionTitle'), 'fa-star-o')
-    addToItems(aboutPagesSuggestions, 'aboutPagesTitle', locale.translation('aboutPagesSuggestionTitle'), null)
-    addToItems(tabSuggestions, 'tabsTitle', locale.translation('tabsSuggestionTitle'), 'fa-external-link')
-    addToItems(searchSuggestions, 'searchTitle', locale.translation('searchSuggestionTitle'), 'fa-search')
-    addToItems(topSiteSuggestions, 'topSiteTitle', locale.translation('topSiteSuggestionTitle'), 'fa-link')
-    const documentHeight = Number.parseInt(window.getComputedStyle(document.querySelector(':root')).getPropertyValue('--navbar-height'), 10)
-    const menuHeight = this.props.menubarVisible ? 30 : 0
-    return <ul className='urlBarSuggestions' style={{
-      maxHeight: document.documentElement.offsetHeight - documentHeight - 2 - menuHeight
-    }}>
-      {items}
-    </ul>
-  }
 
-  onMouseOver (e) {
-    this.updateSuggestions(parseInt(e.target.dataset.index, 10))
+    const list = suggestions.filterSuggestionListByType(this.props.suggestionList)
+
+    addToItems(list.historySuggestions, 'historyTitle', locale.translation('historySuggestionTitle'), 'fa-clock-o')
+    addToItems(list.bookmarkSuggestions, 'bookmarksTitle', locale.translation('bookmarksSuggestionTitle'), 'fa-star-o')
+    addToItems(list.aboutPagesSuggestions, 'aboutPagesTitle', locale.translation('aboutPagesSuggestionTitle'), null)
+    addToItems(list.tabSuggestions, 'tabsTitle', locale.translation('tabsSuggestionTitle'), 'fa-external-link')
+    addToItems(list.searchSuggestions, 'searchTitle', locale.translation('searchSuggestionTitle'), 'fa-search')
+    addToItems(list.topSiteSuggestions, 'topSiteTitle', locale.translation('topSiteSuggestionTitle'), 'fa-link')
+
+    return items
   }
 
-  componentDidMount () {
+  mergeProps (state, dispatchProps, ownProps) {
+    const currentWindow = state.get('currentWindow')
+    const activeFrame = frameStateUtil.getActiveFrame(currentWindow) || Immutable.Map()
+    const urlBar = activeFrame.getIn(['navbar', 'urlbar']) || Immutable.Map()
```

Comment:
just fyi - you can use `activeFrame.getIn(['navbar', 'urlbar'], Immutable.Map())` which is a little safer because it properly handles falsy values when applicable